PMM-3905 Track upstream changes

AlekSi opened the pull request.

Danie committed:
Added pg_statio_user_tables

miguelHx committed:
add support for boolean data types as metrics

This is useful if your database uses true/false for state and want to make prometheus alerts based on that.
Before, booleans were not able to be parsed.  See issue #201

tshohe committed:
add query for v9.1 (#203)

samuelb committed:
Add info about PG_* env vars to README (#199)

robsonpeixoto committed:
Accept constantLabels as parameters

When you area monitoring server from cloud providers the exporter
IP/hostname is not the server IP, so add some constant labels is useful
on dashboards and alerts.

And is very important to use with `--disable-default-metrics` to add
info like `datname` and figure out the database name that you are
monitoring

robsonpeixoto committed:
Speedup docker image build

regeda committed:
multi-server-exporter multi server exporter is introduced

sylr committed:
Add dbname to example to avoid confusion

gsstark committed:
Postgres commits 6e7baa3227 and b06d8e58b5 added a new unit suffix "B"
for memory parameters for bytes.

Commit:
Document --constantLabels

This functionality was added in c225a1e1949e40d387b630683427cb0db259753f
but not documented.

wrouesnel committed:
Improve support code quality.

wrouesnel committed:
Bump Go version.

wrouesnel committed:
Add GIT_API_KEY variable for collecting assets into the repo.

wrouesnel committed:
Fix the errors introduced into smoke-test from the refactor.

wrouesnel committed:
Remove -x from smoke-test.

wrouesnel committed:
Add Postgresql 11 to smoke-tests.

wrouesnel committed:
Update README.md

wrouesnel committed:
Add gh-assets-clone.sh script.

wrouesnel committed:
Update goimport/gofmt of files.

wrouesnel committed:
Fix permissions on gh-asset-clone.

wrouesnel committed:
Add metric change tracking scripts.

wrouesnel committed:
More specific glob for metric files.

wrouesnel committed:
Specify assets branch to get-changes.

wrouesnel committed:
Fix the metric changes script.

wrouesnel committed:
Debug metric push.

wrouesnel committed:
Fix metrics push.

wrouesnel committed:
More debugging.

wrouesnel committed:
Metric push is working.

wrouesnel committed:
Don't fail builds when GIT_API_KEY is not available (on PRs).

We don't need GIT_API_KEY to clone the assets branch, so don't require it.

regeda committed:
fix-param-env-key fix spelling of env parameter

michaelkaye committed:
Update README.md

Allow non-superuser commands to be re-run without (some) errors.

Commit:
Docker file added multistage build to add non-root user running under ID:20001 for image.
Reduces attack surface by not running image as root user

snarlistic committed:
Update the running as non-superuser section. This solves the problem that certain metrics didn't produce a value.

regeda committed:
fix-panic-on-shutdown-server fix panic on shutdown server

regeda committed:
fix-panic-on-shutdown-server added test for const values parsing

arnogeurts-sqills committed:
Add option to automatically discover database names

arnogeurts-sqills committed:
Add documentation for auto-discover-databases

arnogeurts-sqills committed:
Fix build for auto-discover-databases feature

it-percona commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/postgres_exporter?pullRequest=8) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/postgres_exporter?pullRequest=8) before we can accept your contribution.<br/>**1** out of **18** committers have signed the CLA.<br/><br/>:white_check_mark: AlekSi<br/>:x: samuelb<br/>:x: robsonpeixoto<br/>:x: Danie<br/>:x: gsstark<br/>:x: tshohe<br/>:x: wrouesnel<br/>:x: regeda<br/>:x: snarlistic<br/>:x: samsk<br/>:x: asherf<br/>:x: sylr<br/>:x: wolph<br/>:x: arnogeurts-sqills<br/>:x: sakishrist<br/>:x: rkojedzinszky<br/>:x: miguelHx<br/>:x: michaelkaye<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/postgres_exporter?pullRequest=8) it.</sub>

wolph committed:
No need to truncate replication lag to integers

AlekSi committed:
Group and sort imports.

Commit:
Updated the magefile.go to use new version of archiver library

samsk committed:
No version metrics when default metrics disabled

Commit:
Update README.md

add ; for any users that are copy/pasting from the readme

asherf committed:
remove unsupported postgres versions

asherf committed:
Add badge for docker pulls.

wrouesnel committed:
Update travis secrets.

wrouesnel committed:
Update README.md

wrouesnel committed:
Revert "Updated the magefile.go to use new version of archiver library"

This reverts commit 6585e6672fa0fab8599d4db6f48e34df33859b19.

There's been some weird changes added to the upstream archiver library which
break the build. Since this is only used in the build process, I'm in no hurry
to update it.

wrouesnel committed:
Fix smoke-tests to more reliably use pg_isready.

rkojedzinszky committed:
Use application_name column as a label

sakishrist committed:
README: SQL that supports re-execution

Make the SQL work in automated deployments where it will be repeated many times.

DexterHD commented:
Closing due to #9
